.gitignore

```diff
@@ -1,2 +1 @@
 /target
-Cargo.lock
```

Comment:
Can you remind me what the reasoning for going against the Rust defaults / recommendation here is?

I know it doesn't matter to downstream crates, which will just ignore this lock file, but what do we get out of committing it?

Reply:
> I know it doesn't matter to downstream crates, which will just ignore this lock file, but what do we get out of committing it?

From the commit:

> commit lockfile
> This addresses CI bitrot such as the following:
> 
> ```
> error: package `time v0.3.22` cannot be built because it requires rustc 1.65.0 or newer, while the currently active rustc version is 1.62.1
> ```
> 
> Cargo official docs currently recommend against committing a lockfile
> for libraries, but I think they're wrong, and they're considering
> changing this recommendation:
> https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Commit.20lockfiles.20for.20libs.3F/near/323754474
> 
> Applications don't use the library's `Cargo.lock`, but it allows
> stability for CI and documents known-working versions.

to expand a bit on those points:

* Consistency at a given commit when working on the library itself. CI works more reliably (without this I see breakages unrelated to the change you're making) and is more cacheable (though this repo currently doesn't cache anyway), and you can bisect stuff reliably.
* It advertises a known-working set of deps that can be helpful when working on downstream crates too.

Reply:
Ah, thanks. Neither solution is ideal. 😕 If the library can't be built with versions that satisfy Cargo.toml's constraints, it seems like that _should_ break CI to me. Just about the entire point of Cargo.toml is to advertise the working set of deps.

But in any case, I'm fine with it and this PR lgtm!

Reply:
> If the library can't be built with versions that satisfy Cargo.toml's constraints, it seems like that should break CI to me.

We could have one run that uses the checked-in `Cargo.lock` and another that simply deletes it first, so such problems can be isolated. And even run the second set on a schedule to reflect that it's sensitive to external events.

> Just about the entire point of Cargo.toml is to advertise the working set of deps.

Yeah, that's the theory...but there are a lot of caveats, such as MSRV, semver mistakes on the part of dependencies, our use of `bindgen = "*"`, and bugs in general... 🤷

Reply:
(btw, the MSRV one should be fixed by https://github.com/rust-lang/cargo/issues/9930#issue-1002711376 ... provided deps properly set `rust-version`. alternatively, it's fixed by using Rust stable but we might want to avoid using clippy/fmt on a moving target.)